cephfs/cephfs_test.go

```diff
@@ -276,3 +276,75 @@ func TestMdsCommandError(t *testing.T) {
 	assert.NotEqual(t, "", string(info))
 	assert.Contains(t, string(info), "unparseable JSON")
 }
+
+func TestMountWithRoot(t *testing.T) {
+	bMount := fsConnect(t)
+	defer func() {
+		assert.NoError(t, bMount.Unmount())
+		assert.NoError(t, bMount.Release())
+	}()
+
+	dir1 := "/test-mount-with-root"
+	err := bMount.MakeDir(dir1, 0755)
+	assert.NoError(t, err)
+	defer bMount.RemoveDir(dir1)
+
+	sub1 := "/i.was.here"
+	dir2 := dir1 + sub1
+	err = bMount.MakeDir(dir2, 0755)
+	assert.NoError(t, err)
+	defer bMount.RemoveDir(dir2)
+
+	t.Run("withRoot", func(t *testing.T) {
+		mount, err := CreateMount()
+		require.NoError(t, err)
+		require.NotNil(t, mount)
+		defer func() {
+			assert.NoError(t, mount.Unmount())
+			assert.NoError(t, mount.Release())
+		}()
+
+		err = mount.ReadDefaultConfigFile()
+		require.NoError(t, err)
+
+		timeout := time.After(time.Second * 5)
```

Comment:
Are these timeouts really needed? It is probably nice to have them, but they can happen with any test, including usage of rbd images or rados operations...

Reply:
I added the timeouts after I had tests getting stuck to bad / not running ceph instances. I added similar timeouts to the initial setup functions for the other packages too. I only do it once as the initial connection is checked with a `require` and that will fail out of the test function.
Is your issue that this boilerplate distracts from the actual test code?

Reply:
> Is your issue that this boilerplate distracts from the actual test code?

This. Maybe a little. I do not mind that much, but if we have to add this for each network operation it will become a distraction for sure. And maybe a usability issue for go-ceph in general?

Reply:
Fair enough. This was mostly copy and pasted and then adapted from fsConnect. I do feel that fsConnect is a good place to timeout if at least the initial connection can not be established. In this test a connection has been established outside of the t.Run instances to set up our test preconditions.

The secondary issue about these apis all being blocking calls is relevant too but I don't want to bog down this PR for that. Issue #165 touches on this. Let's continue that part of the discussion there or another dedicated issue.

Reply:
I have removed the timeouts from the "inner" parts of the test. It makes the test more readable and the fsConnect function is still called early on and that can provide a good-enough connectivity check.